src/stream.c

```diff
@@ -334,6 +334,9 @@ avifBool avifROStreamReadAndEnforceVersion(avifROStream * stream, uint8_t enforc
 #define AVIF_STREAM_BUFFER_INCREMENT (1024 * 1024)
 static avifResult makeRoom(avifRWStream * stream, size_t size)
 {
+    if (size > SIZE_MAX - stream->offset) {
+        return  AVIF_RESULT_OUT_OF_MEMORY;
+    }
```

Comment:
Yannis: You asked:

> Have you encountered such an integer overflow using the libavif public API?
> If so, it would be good to add a test exercising the fix.

`makeRoom` is called with either a small `size` or the size of a valid memory buffer. We can assume that the size of a valid memory buffer is < `SIZE_MAX / 2`. Since `stream->offset` is <= the size of a valid memory buffer, the sum of `size` and `stream->offset` cannot possibly exceed `SIZE_MAX` under this assumption. So any test that tries to trigger this overflow will first fail in memory allocation.

Having said that, I think it is fine to add this kind of overflow check so that we don't need to perform this kind of analysis.

Reply:
> So any test that tries to trigger this overflow will first fail in memory allocation.

I was thinking of not really allocating the full buffer, but just passing some non-NULL buffer. This should add coverage for the overflow check and fail before triggering an access past the end of the actually allocated buffer.

Reply:
That's a clever idea. Yes, we should be able to make the overflow check fail before accessing the buffer with an invalid size.